include/message_filters/subscriber.h

```diff
@@ -161,61 +136,80 @@ struct message_type <M, false>
 template <typename M>
 using message_type_t = typename message_type<M>::type;
 
-template<class M, class NodeType = rclcpp::Node>
+template<class M>
 class Subscriber 
-: public SubscriberBase<NodeType>
+: public SubscriberBase
 , public SimpleFilter<message_type_t<M>>
 {
 public:
-  typedef std::shared_ptr<NodeType> NodePtr;
   typedef message_type_t<M> MessageType;
   typedef MessageEvent<MessageType const> EventType;
 
   /**
-   * \brief Constructor
+   * \brief Constructor for rclcpp::Node / rclcpp_lifecycle::LifecycleNode
    *
    * See the rclcpp::Node::subscribe() variants for more information on the parameters
    *
-   * \param node The rclcpp::Node::SharedPtr to use to subscribe.
+   * \param node The NodeInterfaces to use to subscribe.
    * \param topic The topic to subscribe to.
    * \param qos (optional) The rmw qos profile to use to subscribe
    */
-  Subscriber(NodePtr node, const std::string& topic, const rmw_qos_profile_t qos = rmw_qos_profile_default)
+  Subscriber(RequiredInterfaces node_interfaces, const std::string& topic,
```

Comment:
When changing the signature here already, maybe include the new storage options as well.
`create_subscription` already supports them. (see [here](https://github.com/ros2/rclcpp/blob/c500695e21dd186981533bb6fb84624e4c4f752c/rclcpp/include/rclcpp/create_subscription.hpp#L205) )